src/data/samplers/modeMedian.js

```diff
@@ -0,0 +1,107 @@
+import d3 from 'd3';
+import {identity, noop} from '../../util/fn';
+
+export default function() {
+
+    var undefinedValue = d3.functor(undefined),
+        number = d3.functor(10),
+        accumulator = noop,
+        field = identity;
+
+    var modeMedian = function(data) {
+
+        if (number() > data.length) {
+            return data;
+        }
+
+        var minMax = getGlobalMinMax(data);
+        var buckets = getBuckets(data);
+
+        var subsampledData = buckets.map(function(bucket, i) {
+
+            var frequencies = {};
+            var mostFrequent;
+            var mostFrequentIndex;
+            var singleMostFrequent = true;
+
+            for (var j = 0; j < bucket.length; j++) {
+                var item = field(bucket[j]);
+                if (item === minMax[0] || item === minMax[1]) {
+                    return bucket[j];
+                }
+
+                if (frequencies[item] === undefined) {
+                    frequencies[item] = 0;
+                }
+                frequencies[item]++;
+
+                if (frequencies[item] > frequencies[mostFrequent] || mostFrequent === undefined) {
+                    mostFrequent = item;
+                    mostFrequentIndex = j;
+                    singleMostFrequent = true;
+                } else if (frequencies[item] === frequencies[mostFrequent]) {
+                    singleMostFrequent = false;
+                }
+            }
+
+            if (singleMostFrequent) {
+                return bucket[mostFrequentIndex];
+            } else {
+                return bucket[Math.floor(bucket.length / 2)];
+            }
+        });
+
+        // First and last data points are their own buckets.
+        return [].concat(data[0], subsampledData, data[data.length - 1]);
+    };
+
+    var getGlobalMinMax = function(data) {
```

Comment:
Can you change these from anonymous functions into function declarations? see this file for the pattern:

https://github.com/ScottLogic/d3fc/blob/master/src/series/bar.js

And this page for more info:

http://stackoverflow.com/questions/336859/var-functionname-function-vs-function-functionname